dpnp/backend/extensions/ufunc/elementwise_functions/nan_to_num.cpp

```diff
@@ -0,0 +1,408 @@
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <sycl/sycl.hpp>
+
+#include "dpctl4pybind11.hpp"
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "kernels/elementwise_functions/nan_to_num.hpp"
+
+#include "../../elementwise_functions/simplify_iteration_space.hpp"
+
+// dpctl tensor headers
+#include "utils/memory_overlap.hpp"
+#include "utils/offset_utils.hpp"
+#include "utils/output_validation.hpp"
+#include "utils/sycl_alloc_utils.hpp"
+#include "utils/type_dispatch.hpp"
+#include "utils/type_utils.hpp"
+
+namespace py = pybind11;
+namespace td_ns = dpctl::tensor::type_dispatch;
+
+// declare pybind11 wrappers in py_internal namespace
+namespace dpnp::extensions::ufunc
+{
+
+namespace impl
+{
+
+template <typename T>
+struct value_type_of
+{
+    using type = T;
+};
+
+template <typename T>
+struct value_type_of<std::complex<T>>
+{
+    using type = T;
+};
+
+template <typename T>
+using value_type_of_t = typename value_type_of<T>::type;
+
+typedef sycl::event (*nan_to_num_fn_ptr_t)(sycl::queue &,
+                                           int,
+                                           std::size_t,
+                                           const py::ssize_t *,
+                                           const py::object &,
+                                           const py::object &,
+                                           const py::object &,
+                                           const char *,
+                                           py::ssize_t,
+                                           char *,
+                                           py::ssize_t,
+                                           const std::vector<sycl::event> &);
+
+template <typename T>
+sycl::event nan_to_num_strided_call(sycl::queue &exec_q,
+                                    int nd,
+                                    std::size_t nelems,
+                                    const py::ssize_t *shape_strides,
+                                    const py::object &py_nan,
+                                    const py::object &py_posinf,
+                                    const py::object &py_neginf,
+                                    const char *arg_p,
+                                    py::ssize_t arg_offset,
+                                    char *dst_p,
+                                    py::ssize_t dst_offset,
+                                    const std::vector<sycl::event> &depends)
+{
+    using dpctl::tensor::type_utils::is_complex_v;
+    using scT = std::conditional_t<is_complex_v<T>, value_type_of_t<T>, T>;
+
+    const scT nan_v = py::cast<const scT>(py_nan);
+    const scT posinf_v = py::cast<const scT>(py_posinf);
+    const scT neginf_v = py::cast<const scT>(py_neginf);
+
+    using dpnp::kernels::nan_to_num::nan_to_num_impl;
+    sycl::event to_num_ev = nan_to_num_impl<T, scT>(
```

Comment:
It'd be good to add `strided` suffix here also:
```suggestion
    sycl::event to_num_ev = nan_to_num_strided_impl<T, scT>(
```

Reply:
Good point

Reply:
I've added this change